cirq-core/cirq/sim/classical_simulator.py

```diff
@@ -12,96 +12,134 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Dict
-from collections import defaultdict
-from cirq.sim.simulator import SimulatesSamples
-from cirq import ops, protocols
-from cirq.study.resolver import ParamResolver
-from cirq.circuits.circuit import AbstractCircuit
+
+from typing import Dict, Generic, Any, Sequence, TYPE_CHECKING, List
+from cirq import ops, qis
+from cirq.value import big_endian_int_to_bits
 from cirq.ops.raw_types import Qid
+from cirq import sim
+from cirq.sim.simulation_state import TSimulationState, SimulationState
 import numpy as np
 
+if TYPE_CHECKING:
+    import cirq
+
+
+class ClassicalBasisState(qis.QuantumStateRepresentation):
+    def __init__(self, initial_state: List[int]):
+        self.basis = initial_state
+
+    def copy(self, deep_copy_buffers: bool = True) -> 'ClassicalBasisState':
+        basis_copy = self.basis if deep_copy_buffers else self.basis.copy()
+        return ClassicalBasisState(basis_copy)
+
+    def measure(self, axes: Sequence[int], seed: 'cirq.RANDOM_STATE_OR_SEED_LIKE' = None):
+        return [self.basis[i] for i in axes]
+
+
+class ClassicalBasisSimState(SimulationState[ClassicalBasisState]):
+    def __init__(self, initial_state, qubits, classical_data):
+        state = ClassicalBasisState(big_endian_int_to_bits(initial_state, bit_count=len(qubits)))
```

Comment:
>> big_endian_int_to_bits(initial_state, bit_count=len(qubits))

this assumes `initial_state` is always an int which is not true. please handle the other cases and document well -in the docstring- that integer initial state gets converted in this way.